protocols/request-response/src/json.rs

```diff
@@ -54,26 +54,41 @@ mod codec {
     use libp2p_swarm::StreamProtocol;
     use serde::{de::DeserializeOwned, Serialize};
 
-    /// Max request size in bytes
-    const REQUEST_SIZE_MAXIMUM: u64 = 1024 * 1024;
-    /// Max response size in bytes
-    const RESPONSE_SIZE_MAXIMUM: u64 = 10 * 1024 * 1024;
-
     pub struct Codec<Req, Resp> {
+        /// Max request size in bytes
+        request_size_maximum: u64,
+        /// Max response size in bytes
+        response_size_maximum: u64,
         phantom: PhantomData<(Req, Resp)>,
     }
 
     impl<Req, Resp> Default for Codec<Req, Resp> {
         fn default() -> Self {
             Codec {
+                request_size_maximum: 1024 * 1024,
+                response_size_maximum: 10 * 1024 * 1024,
                 phantom: PhantomData,
             }
         }
     }
 
     impl<Req, Resp> Clone for Codec<Req, Resp> {
         fn clone(&self) -> Self {
-            Self::default()
+            Self {
+                request_size_maximum: self.request_size_maximum,
+                response_size_maximum: self.response_size_maximum,
+                phantom: self.phantom,
+            }
+        }
+    }
+
+    impl<Req, Resp> Codec<Req, Resp> {
+        pub fn new(request_size_maximum: u64, response_size_maximum: u64) -> Self {
+            Self {
+                request_size_maximum,
+                response_size_maximum,
+                ..Default::default()
+            }
         }
```

Comment:
I would prefer if these are not parameters in the `new` function, since most users are fine with the default values.
I know we still have `Codec::default`, but from my own experience I think many user will intuitively use the `new` constructor and then be forced to think about these number.

I would instead propose to either:
- have a `Config` parameter with methods `set_request_size_maximum` `set_response_size_maximum` 
- have the methods `set_request_size_maximum` `set_response_size_maximum` directly on `Codec`

Reply:
Make sense. I pushed up the change for it.